eos/nonleptonic-amplitudes/topological-amplitudes.cc

```diff
@@ -0,0 +1,98 @@
+/* vim: set sw=4 sts=4 et foldmethod=syntax : */
+
+/*
+ * Copyright (c) 2024 Méril Reboud
+ *
+ * This file is part of the EOS project. EOS is free software;
+ * you can redistribute it and/or modify it under the terms of the GNU General
+ * Public License version 2, as published by the Free Software Foundation.
+ *
+ * EOS is distributed in the hope that it will be useful, but WITHOUT ANY
+ * WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE.  See the GNU General Public License for more
+ * details.
+ *
+ * You should have received a copy of the GNU General Public License along with
+ * this program; if not, write to the Free Software Foundation, Inc., 59 Temple
+ * Place, Suite 330, Boston, MA  02111-1307  USA
+ */
+
+
+#include <eos/nonleptonic-amplitudes/topological-amplitudes.hh>
+#include <eos/maths/power-of.hh>
+#include <eos/utils/options-impl.hh>
+
+#include <map>
+
+namespace eos
+{
+    using std::sqrt;
+
+    NonleptonicAmplitudes<PToPP> *
+    TopologicalRepresentation<PToPP>::make(const Parameters & p, const Options & o)
+    {
+        return new TopologicalRepresentation<PToPP>(p, o);
+    }
+
+
+    const std::vector<OptionSpecification>
+    TopologicalRepresentation<PToPP>::options
+    {
+        Model::option_specification(),
+        { "q", { "u", "d", "s" }, "d" },
+        { "m1", { "pi^0", "pi^+", "K_d", "K_u", "eta_q", "eta_q'", "eta_s", "eta_s'" }, "pi^0" },
+        { "m2", { "pi^0", "pi^+", "K_d", "K_u", "eta_q", "eta_q'", "eta_s", "eta_s'" }, "pi^0" },
+    };
+
+    complex<double>
+    TopologicalRepresentation<PToPP>::tree_amplitude() const
+    {
+        complex<double> T_tda = 0.0;
+        for (unsigned i = 0; i < 3; i++)
+        {
```

Comment:
To avoid evaluating all the parameters at each iteration, use `auto T = this->T()` here.